sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ConstantPropagationSuite.scala

```diff
@@ -37,10 +38,11 @@ class ConstantPropagationSuite extends PlanTest {
         Batch("ConstantPropagation", FixedPoint(10),
           ConstantPropagation,
           ConstantFolding,
-          BooleanSimplification) :: Nil
+          BooleanSimplification,
+          PruneFilters) :: Nil
```

Comment:
my question here was: why do we need to add this rule? I see it can avoid having filters containing only `true` or `false` for instance, but is it really needed to test the constant propagation? I mean, I expect the output to contain useless filters in the plan, but that's fine: the goal of the tests is not to get the best plan, but the correct one adn the less rules we have for testing a specific one, the easier to debugger IMHO.

Reply:
Thanks, I see you point now. I've removed these additions and changed the expected filters a bit.